### PR TITLE
Install profile

### PR DESCRIPTION
**Pull request by rosiel**
This PR updates the demo box to be built using islandora-sandbox, which uses the islandora_install_profile_demo. Part of https://github.com/Islandora-Devops/isle-dc/pull/160

### CODE REVIEW

**Comment by rosiel** [comment]
Once sandbox (https://github.com/Islandora-Devops/islandora-sandbox/pull/5) gets merged, we could update the instructions for which commit to pull down. At the very least, I'd be in favour of updating it to pull from islandora-devops not dannylamb.

File: drupal/rootfs/usr/local/bin/install-drupal-site.sh [hunk 1]
```diff
@@ -164,6 +164,7 @@ function main {
         protocol=pgsql
     fi
     echo "Installing site."
+    echo "-n si ${DRUSH_ARGS} --db-url=${protocol}://${DB_USER}:${DB_PASSWORD}@${HOST}:${PORT}/${DB_NAME}"
```
Comment (nigelgbanks) [inline drupal/rootfs/usr/local/bin/install-drupal-site.sh:167]: Same as above, just concerned about leaking passwords.

File: drupal/rootfs/etc/islandora/utilities.sh [hunk 2]
```diff
@@ -246,6 +246,23 @@ function install_site {
         chmod a=rwx "${site_directory}/settings.php"
     fi
 
+    echo "--driver ${driver}"
+    echo "--host ${host}"
+    echo "--port ${port}"
+    echo "--dbuser ${user}"
+    echo "--dbpassword ${password}"
```
Comment (nigelgbanks) [inline drupal/rootfs/etc/islandora/utilities.sh:253]: We probably shouldn't print passwords to std-out

**Review by nigelgbanks — COMMENTED** [review]
Aside from the password concern, everything else looks good. I'll try doing some testing tomorrow.

**Comment by rosiel** [comment]
Changed the repos...

I'm not sure why it's printing passwords... is it just letting you know that the password for the container you just built is the thing you set it as in the file in buildkit?

File: drupal/rootfs/etc/islandora/utilities.sh (hunk 2 above)
Comment (rosiel) [reply drupal/rootfs/etc/islandora/utilities.sh:253]: Any suggestions? (I'm at a loss, I don't have enough Docker management experience to know what the risk is or how to do it better.)

**Comment by nigelgbanks** [comment]
Tried testing this today with `isle-dc` though it looks like some aspect of Drush has changed?

**Comment by nigelgbanks** [comment]
> I'm not sure why it's printing passwords... is it just letting you know that the password for the container you just built is the thing you set it as in the file in buildkit?

`echo` goes to `stdout` and that all gets captured by Docker logs, I'll just remove those lines as part of testing. The user of isle-dc can change the file on disk for the password, or there could be an `isle-dc` command or last step that prints passwords out. It's just that it's not only `isle-dc` that consumes these images some people are running in ECS/Kubernetes etc and for them they wouldn't want their administrative passwords going into a log aggregator.

**Comment by nigelgbanks** [comment]
Sorry for taking a while to get around to this @rosiel, I've raised a pull request that fixes the merge conflict with main and brings in the changes from main into this branch. Also remove the spots where passwords would be printed out.

https://github.com/Islandora-Devops/isle-buildkit/pull/177

I really tried to test it with isle-dc in particular with this branch: https://github.com/Islandora-Devops/isle-dc/pull/160

But it was failing and I wasn't able to sort out why. So I've not actually been able to test the changes for the install profile.

**Comment by rosiel** [comment]
@nigelgbanks thanks for the review! The demo/Dockerfile is still conflicting, but we *want* to remove all those extra commands from that Dockerfile, right?

**Comment by nigelgbanks** [comment]
@rosiel Yup, I just merged main to get rid of them. I tested with https://github.com/Islandora-Devops/isle-dc/tree/install-profile but it doesn't seem to work with `make demo`. 

[log.txt](https://github.com/Islandora-Devops/isle-buildkit/files/7864575/log.txt)

I also tried `make demo-install-profile` which finished running but the site was missing content and kinda broken.

**Comment by rosiel** [comment]
missing content: yes, that's the plan for the moment because we dont' have a way of putting default content in. @Natkeeran was working on a method using Workbench. 
Kinda broken: yes. That's not buildkit's fault as far as I can see. 

`make demo` doesn't work? oh dear. I'm not sure what to do about that. @dannylamb ?

**Comment by nigelgbanks** [comment]
Just to put a note here, I'm working on automating the deployment of the sandbox and will update this with the output of that work once complete.

**Comment by misilot** [comment]
With the merge of https://github.com/Islandora-Devops/isle-dc/pull/160 does this need to be merged, and a new version released?

**Comment by nigelgbanks** [comment]
@misilot you can reference `211bfffb9679221dee76562b699d7ae61cbabc86` for now, one this has been confirmed to work it will be merged.

**Comment by misilot** [comment]
@rosiel any chance you could rebase this PR  again? We ran into an issue with our system, so looking to test this again and stand up a new system.

**Comment by misilot** [comment]
@nigelgbanks do we need to update the TAG in isle-dc with this version? Since it is referencing the old (before rebasing) version currently

**Comment by misilot** [comment]
I am seeing the following when trying to build with  `TAG=211bfffb9679221dee76562b699d7ae61cbabc86`

@DonRichards I see it here https://hub.docker.com/layers/drupal/islandora/drupal/211bfffb9679221dee76562b699d7ae61cbabc86/images/sha256-66d57d2bd486d8034d07d3b2ff9fc54e74ab4c455e3fa707cf668b61dfa7ee4e?context=explore (At least I assume this is the same?)

```
 [error]  Config search_api.server.default_solr_server does not exist
 [error]  Config search_api.server.default_solr_server does not exist
make[1]: [Makefile:155: update-config-from-environment] Error 1 (ignored)
WARNING: Some services (activemq, alpaca, blazegraph, cantaloupe, crayfits, drupal, fcrepo, fits, homarus, houdini, hypercube, mariadb, matomo, milliner, recast, solr, traefik) use the 'deploy' key, which will be ignored. Compose does not support 'deploy' configuration - use `docker stack deploy` to deploy to a swarm.

In SolrCommandHelper.php line 119:

  Unknown server default_solr_server




ERROR: Could not generate SOLR config.zip!
In Drupal, check Configuration -> Search API -> SOLR Server, and use the
+ Get config.zip option which should give you information into the actual error.


make[1]: *** [Makefile:167: solr-cores] Error 1
make[1]: Leaving directory '/klib/app/docker-compose/isle-dc'
make: *** [Makefile:390: local-install-profile] Error 2
```

**Comment by nigelgbanks** [comment]
I've rebased and raised https://github.com/Islandora-Devops/isle-buildkit/pull/209 using the latest commit from the sandbox in hopes to solve the `make demo` issues on https://github.com/Islandora-Devops/isle-dc/pull/248 to get `isle-dc` into a working state.

**Pull request closed without merging**